src/finch/interface/lazy.py

```diff
@@ -327,7 +327,64 @@ def prod(arr: LazyTensor, dims) -> LazyTensor:
     .. versionchanged:: 2023.12
        Required the function to return a floating-point array having the same data type as the input array when provided a floating-point array.
     """
-    return reduce(operator.mul, arr, dims, arr.fill_value)
+    return reduce(operator.mul, arr, axis=dims, init=arr.fill_value)
 
 def multiply(x1: LazyTensor, x2: LazyTensor) -> LazyTensor:
-    return elementwise(operator.mul, x1, x2)
\ No newline at end of file
+    return elementwise(operator.mul, x1, x2)
+
+
+def sum(arr: LazyTensor, dims) -> LazyTensor:
```

Comment:
Does not conform to the function signature for sum: https://data-apis.org/array-api/2021.12/API_specification/generated/array_api.sum.html